src/repr/src/row/encoding2.rs

```diff
@@ -183,10 +190,25 @@ impl DatumColumnEncoder {
             (DatumColumnEncoder::I64(builder), Datum::Int64(val)) => builder.append_value(val),
             (DatumColumnEncoder::F32(builder), Datum::Float32(val)) => builder.append_value(*val),
             (DatumColumnEncoder::F64(builder), Datum::Float64(val)) => builder.append_value(*val),
-            (DatumColumnEncoder::Numeric(builder), d @ Datum::Numeric(_)) => {
-                let proto = ProtoDatum::from(d);
-                let bytes = proto.encode_to_vec();
-                builder.append_value(&bytes);
+            (
+                DatumColumnEncoder::Numeric {
+                    approx_values,
+                    binary_values,
+                },
+                Datum::Numeric(val),
+            ) => {
+                let mut ctx = Context::default();
+                let float_approx = ctx.try_into_f64(val.0).unwrap_or_else(|_| {
```

Comment:
Oh, this is interesting! I take it that the only case where converting to f64 fails is when the numeric is out of range?

Reply:
Exactly! Reading the code I don't think this should ever happen, but adding this just to be safe